### PR TITLE
Adapt Animation events to new structure

### PR DESCRIPTION
**Pull request by teoli2003**
Per the new events guideline (Part of https://github.com/mdn/browser-compat-data/issues/14578).

Content work: https://github.com/mdn/content/pull/12882

### CODE REVIEW

File: api/Animation.json [hunk 1]
```diff
@@ -462,16 +417,20 @@
           }
         }
       },
-      "oncancel": {
+      "finish_event": {
         "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Animation/oncancel",
-          "spec_url": "https://drafts.csswg.org/web-animations-1/#dom-animation-oncancel",
+          "description": "<code>finish</code> event",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Animation/onfinish",
```
Comment (Elchi3) [inline api/Animation.json:423]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Animation/finish_event",
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Almost :)

**Comment by teoli2003** [comment]
Sorry I forgot the mdn_url !

**Review by Elchi3 — APPROVED** [review]
No worries! Nice work! 👍

**Pull request merged**